src/components/NavBar/MenuSwitch/index.tsx

```diff
@@ -11,34 +12,94 @@ import {
   LayersCategoryType,
   LayerType,
   TableType,
 } from '../../../config/types';
-import { removeLayer } from '../../../context/mapStateSlice';
+import { addLayer, removeLayer } from '../../../context/mapStateSlice';
 import { loadTable } from '../../../context/tableStateSlice';
-import { layersSelector } from '../../../context/mapStateSlice/selectors';
+import {
+  layersSelector,
+  mapSelector,
+} from '../../../context/mapStateSlice/selectors';
 import { useUrlHistory } from '../../../utils/url-utils';
+import {
+  getDisplayBoundaryLayers,
+  getBoundaryLayerSingleton,
+  LayerDefinitions,
+} from '../../../config/utils';
+import { isLayerOnView } from '../../../utils/map-utils';
 
 function MenuSwitch({ classes, title, layers, tables }: MenuSwitchProps) {
   const selectedLayers = useSelector(layersSelector);
+  const map = useSelector(mapSelector);
   const dispatch = useDispatch();
   const { updateHistory, removeKeyFromUrl } = useUrlHistory();
 
+  const safeDispatchAddLayer = (
+    _map: MapBoxMap | undefined,
+    layer: LayerType,
+  ) => {
+    if (!isLayerOnView(_map, layer.id)) {
+      dispatch(addLayer(layer));
+    }
+  };
+
+  const safeDispatchRemoveLayer = (
+    _map: MapBoxMap | undefined,
+    layer: LayerType,
+  ) => {
+    if (isLayerOnView(_map, layer.id)) {
+      dispatch(removeLayer(layer));
+    }
+  };
+
   const toggleLayerValue = (layer: LayerType) => (
     event: React.ChangeEvent<HTMLInputElement>,
   ) => {
+    const ADMIN_LEVEL_DATA_LAYER_KEY = 'admin_level_data';
     const { checked } = event.target;
 
     const urlLayerKey =
-      layer.type === 'admin_level_data' ? 'baselineLayerId' : 'hazardLayerId';
+      layer.type === ADMIN_LEVEL_DATA_LAYER_KEY
+        ? 'baselineLayerId'
+        : 'hazardLayerId';
 
     if (checked) {
       updateHistory(urlLayerKey, layer.id);
+
+      const defaultBoundary = getBoundaryLayerSingleton();
+      if (!('boundary' in layer) && layer.type === ADMIN_LEVEL_DATA_LAYER_KEY) {
+        safeDispatchAddLayer(map, defaultBoundary);
+      }
     } else {
       removeKeyFromUrl(urlLayerKey);
-
-      dispatch(removeLayer(layer));
+      safeDispatchRemoveLayer(map, layer);
+
+      // For admin boundary layers with boundary property
+      // we have to de-activate the unique boundary and re-activate
+      // default boundaries
+      if ('boundary' in layer) {
+        const boundaryId = layer.boundary || '';
+
+        if (Object.keys(LayerDefinitions).includes(boundaryId)) {
+          const displayBoundaryLayers = getDisplayBoundaryLayers();
+          const uniqueBoundaryLayer = LayerDefinitions[boundaryId as LayerKey];
+
+          if (
+            !displayBoundaryLayers
+              .map(l => l.id)
+              .includes(uniqueBoundaryLayer.id)
+          ) {
+            safeDispatchRemoveLayer(map, uniqueBoundaryLayer);
+          }
+
+          displayBoundaryLayers.forEach(l => {
+            safeDispatchAddLayer(map, l);
+          });
```

Comment:
I am not sure I understand what we are trying to achieve here? Can you explicit and/or simplify the logic?